students/mattmaeda/session11/iterator_1.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+
+"""
+Simple iterator examples
+"""
+
+
+class IterateMe_1(object):
+    """
+    About as simple an iterator as you can get:
+
+    returns the sequence of numbers from zero to 4
+    ( like range(4) )
+    """
+    def __init__(self, stop=5):
+        self.current = -1
+        self.stop = stop
+    def __iter__(self):
+        self.current = -1
+        return self
+    def __next__(self):
+        self.current += 1
+        if self.current < self.stop:
+            return self.current
+        else:
+            raise StopIteration
+
+
+class IterateMe_2(IterateMe_1):
+    """ Override IterateMe_1 and give it range like behavior
+    """
+    def __init__(self, start, stop=None, step=1):
+        real_stop = None
+
+        if stop is None:
+            real_stop = start
+        else:
+            real_stop = stop
+
+        super(IterateMe_2, self).__init__(real_stop)
```

Comment:
since the __init__ signature is changed, I'd not bother calling super here, and just initialize everything you need.